client/alloc_runner.go

```diff
@@ -142,9 +143,16 @@ func (r *AllocRunner) RestoreState() error {
 		// Mark the task as restored.
 		r.restored[name] = struct{}{}
 
+		td, ok := r.allocDir.TaskDirs[name]
+		if !ok {
+			err := fmt.Errorf("failed to find task dir metadata for alloc %q task %q",
+				r.alloc.ID, name)
+			r.logger.Printf("[ERR] client: %v", err)
+			mErr.Errors = append(mErr.Errors, err)
```

Comment:
We should error out and not do anything else if we can't find the task directory from the alloc dir struct. You will be passing nil otherwise to the TaskRunner.